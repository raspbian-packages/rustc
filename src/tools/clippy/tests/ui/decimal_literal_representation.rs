


#[warn(decimal_literal_representation)]
#[allow(unused_variables)]
fn main() {
    // Hex:      7F,  80, 100, 1FF,  800,  FFA,   F0F3,     7F0F_F00D
    let good = (127, 128, 256, 511, 2048, 4090, 61_683, 2_131_750_925);
    let bad = (        // Hex:
        4096,          // 0x1000
        16_371,        // 0x3FF3
        32_773,        // 0x8005
        65_280,        // 0xFF00
        2_131_750_927, // 0x7F0F_F00F
        2_147_483_647, // 0x7FFF_FFFF
        4_042_322_160, // 0xF0F0_F0F0
    );
}
