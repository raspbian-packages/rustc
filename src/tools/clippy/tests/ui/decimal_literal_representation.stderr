error: integer literal has a better hexadecimal representation
  --> $DIR/decimal_literal_representation.rs:10:9
   |
10 |         4096,          // 0x1000
   |         ^^^^
   |
   = note: `-D decimal-literal-representation` implied by `-D warnings`
   = help: consider: 0x1000

error: integer literal has a better hexadecimal representation
  --> $DIR/decimal_literal_representation.rs:11:9
   |
11 |         16_371,        // 0x3FF3
   |         ^^^^^^
   |
   = help: consider: 0x3FF3

error: integer literal has a better hexadecimal representation
  --> $DIR/decimal_literal_representation.rs:12:9
   |
12 |         32_773,        // 0x8005
   |         ^^^^^^
   |
   = help: consider: 0x8005

error: integer literal has a better hexadecimal representation
  --> $DIR/decimal_literal_representation.rs:13:9
   |
13 |         65_280,        // 0xFF00
   |         ^^^^^^
   |
   = help: consider: 0xFF00

error: integer literal has a better hexadecimal representation
  --> $DIR/decimal_literal_representation.rs:14:9
   |
14 |         2_131_750_927, // 0x7F0F_F00F
   |         ^^^^^^^^^^^^^
   |
   = help: consider: 0x7F0F_F00F

error: integer literal has a better hexadecimal representation
  --> $DIR/decimal_literal_representation.rs:15:9
   |
15 |         2_147_483_647, // 0x7FFF_FFFF
   |         ^^^^^^^^^^^^^
   |
   = help: consider: 0x7FFF_FFFF

error: integer literal has a better hexadecimal representation
  --> $DIR/decimal_literal_representation.rs:16:9
   |
16 |         4_042_322_160, // 0xF0F0_F0F0
   |         ^^^^^^^^^^^^^
   |
   = help: consider: 0xF0F0_F0F0

error: aborting due to 7 previous errors

